tests/project/_internal/test_mbed_program.py

```diff
@@ -90,7 +90,6 @@ def test_from_existing_raises_if_path_is_not_a_program(self, fs):
     def test_from_existing_raises_if_program_files_missing(self, fs):
         fs_root = pathlib.Path(fs, "foo")
         fs_root.mkdir()
-        (fs_root / ".mbed").touch()
```

Comment:
should we `(fs_root /  "mbed-os.lib").touch()` do this here?

Reply:
I'm not sure what you mean. We've removed all references to `.mbed` so we don't need to test it here.

Reply:
If you see `test_from_new_local_dir_raises_if_path_is_existing_program`  was replaced `(program_root / ".mbed").touch()` ... unless I got `test_from_existing_raises_if_program_files_missing` test wrong?

Reply:
Sorry, I see what you mean now... So before we checked for `.mbed` and `mbed-os.lib`, and this test was just checking `mbed-os.lib`, so we had to touch `.mbed`, so it didn't raise in the wrong place. Now `.mbed` isn't checked, so we raise in the wrong place (in _find_program_root, instead of MbedProgram.from_existing). I'll rework it a bit so it's more obvious. Good catch.

Reply:
I've removed some duplicated error handling in `MbedProgramFiles` and deleted this test.